### PR TITLE
WIP: add basemeasure for models

### PR DESCRIPTION
**Pull request by mschauer**

### CODE REVIEW

File: src/primitives/basemeasure.jl [hunk 1]
```diff
@@ -0,0 +1,46 @@
+
+export basemeasure
+
+import MeasureTheory
+
+function MeasureTheory.basemeasure(c::ConditionalModel{A,B,M}, x=NamedTuple()) where {A,B,M}
+    _basemeasure(M, Model(c), argvals(c), observations(c), x)
```
Comment (mschauer) [inline src/primitives/basemeasure.jl:7]: Discuss if this is the right signature? One could make it a function of a Model too, right?

Comment (cscherrer) [reply src/primitives/basemeasure.jl:7]: Let's assume not, it will often depend on the arguments

File: src/primitives/basemeasure.jl [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        # proc(_m, st :: Sample)     = :(_ℓ += basemeasure($(st.rhs), $(st.x)))
+        proc(_m, st :: Return)     = nothing
+        proc(_m, st :: LineNumber) = nothing
+        function proc(_m, st :: Sample)
+            x = st.x
+            xname = QuoteNode(x)
+            rhs = st.rhs
+            @q begin
+                _bm = merge(_bm, NamedTuple{($xname,)}((basemeasure($rhs),)))
+                $x = Soss.predict($rhs, $x)
```
Comment (cscherrer) [inline src/primitives/basemeasure.jl:26]: Maybe we don't need this? I think we ought to be able to do something like
```julia
$x = getproperty(_data, $xname)
```

File: src/primitives/basemeasure.jl [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+end
+
+export sourceBasemeasure
+
+sourceBasemeasure(m::AbstractModel) = sourceBasemeasure()(Model(m))
+
+function sourceBasemeasure()
+    function(_m::Model)
+        proc(_m, st :: Assign)     = :($(st.x) = $(st.rhs))
+        # proc(_m, st :: Sample)     = :(_ℓ += basemeasure($(st.rhs), $(st.x)))
```
Comment (cscherrer) [inline src/primitives/basemeasure.jl:17]:
```suggestion
```

File: src/primitives/basemeasure.jl [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+            xname = QuoteNode(x)
+            rhs = st.rhs
+            @q begin
+                _bm = merge(_bm, NamedTuple{($xname,)}((basemeasure($rhs),)))
+                $x = Soss.predict($rhs, $x)
+            end
+        end
+
+        wrap(kernel) = @q begin
+            #_bm = ProductMeasure(())
```
Comment (cscherrer) [inline src/primitives/basemeasure.jl:31]:
```suggestion
```

**Comment by mschauer** [comment]
@sebasgraz

**Comment by mschauer** [comment]
@sebagraz

**Review by cscherrer — APPROVED** [review]

**Comment by mschauer** [comment]
Runs locally with master.

**Pull request merged**